index-template-mapping.json

```diff
@@ -8,15 +8,16 @@
     }
   },
   "mappings": {
-    "_source": { "enabled": true },
-    "properties": {
-      "@timestamp": { "type": "date" },
-      "@version": { "type": "keyword" },
-      "message": { "type": "text", "index": true },
-      "severity": { "type": "keyword", "index": true },
-      "fields": {
-        "dynamic": true,
-        "properties": { }
+    "winston-elasticsearch-test-log-type": {
```

Comment:
I guess this is the name of the type that will then appear in Kibana? 
What about `_source` why have you removed that?

Reply:
Ah I think I realize now why the template mapping didn't work. I use ES 6. And you probably use ES 7... ES 7 removed mapping type and the template json structure changed for ES 7. So the existing template json works on ES 7 and my template works on ES 6.
I will re-implement my fix to have two templates and auto-detect the ES version before the tests run and use the right one (it's just for the tests to work I believe.. right?).

And about the template name appearing in kibana.. yes it will appear as the _type field of each document. ES documentation uses "_doc" as type name when you don't want to give a fancy type name. I'll do that as well.

Reply:
And about _source..
1. source is enabled by default anyway. So it's redundant to mention it. But we can add it if you want to be explicit about it.
2. the syntax should've been "_source": { "enabled": true } and not "_source": true
ES 6 throws an error with that. (I think it's not valid in ES 7 either from the docs -> https://www.elastic.co/guide/en/elasticsearch/reference/7.8/mapping-source-field.html)

Reply:
ok, agreed